### PR TITLE
[docs] Setup a new next.js application for documentation

### PR DESCRIPTION
**Pull request by brijeshb42**
This app only uses Pigment and some CSS module for styling and will serve as an internal dogfood for us.

The styling and most of the mdx related logic has been taken from the Base UI docs.
Similarly, it uses the app router with static export to build the whole docs statically that we can then deploy on Netlify.

Preview - https://deploy-preview-299--pigment-css.netlify.app/

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/pigment-css/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: docs/data/getting-started/overview/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+description: Get started with Pigment CSS, a zero-runtime CSS-in-JS library.
+---
+
+# Quick start
+
+<Description />
+
+## Installation
+
+Pigment CSS has two category of packages. First one is to be imported and used in your source code. This includes the public API of the package. Second category is to be imported in your bundler config file to configure and actually be able to use Pigment CSS. We currently support [`Next.js`](https://nextjs.org/) (no Turbopack yet), [`Vite`](https://vite.dev/) and [`Webpack`](https://webpack.js.org/).
```
Comment (mnajdova) [inline docs/data/getting-started/overview/overview.mdx:12]:
```suggestion
Pigment CSS has two category of packages.
First one is to be imported and used in your source code.
This includes the public API of the package.
Second category is to be imported in your bundler config file to configure and actually be able to use Pigment CSS.
We currently support [`Next.js`](https://nextjs.org/) (no Turbopack yet), [`Vite`](https://vite.dev/) and [`Webpack`](https://webpack.js.org/).
```

Not important for this PR, but putting each sentence in a new line, makes the review so much easier. This is the convention we use in the Material UI's docs

File: docs/next.config.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import type { NextConfig } from 'next';
+import { withPigment, extendTheme } from '@pigment-css/nextjs-plugin';
+import path from 'path';
+import { theme as baseTheme } from './src/theme';
+
+const DATA_DIR = path.join(process.cwd(), 'data');
+
+const nextConfig: NextConfig = {
+  /* config options here */
```
Comment (mnajdova) [inline docs/next.config.ts:9]:
```suggestion
```

**Comment by brijeshb42** [comment]
@mnajdova 

1. Done
2. We can revisit this separately. I just copied over the current svg from README.
3. It is disabled right now since there is just one version. I copied this over from Base UI. We could probably comment it out now and add it back when we actually need it.

I'll merge this and do more refinements (point 2 and 3) in separate PRs.

File: docs/data/getting-started/overview/overview.mdx (hunk 1 above)
Comment (brijeshb42) [reply docs/data/getting-started/overview/overview.mdx:12]: Will do in separate PRs. This wasn't even real content.

**Comment by mnajdova** [comment]
Sounds good 👍

File: docs/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  rules: {
+    'react/prop-types': 'off',
+    'react/react-in-jsx-scope': 'off',
+    'react/no-unknown-property': ['error', { ignore: ['sx'] }],
+    'import/extensions': [
+      'error',
```
Comment (oliviertassinari) [inline docs/.eslintrc.js:7]: Config needs to move to the root folder eslint config file to prepare Eslint v9 upgrade.

**Comment by oliviertassinari** [comment]
I wired it to https://pigment-css.com/ with `docs-v1` branch for production.

**Comment by mnajdova** [comment]
> I wired it to https://pigment-css.com/ with docs-v1 branch for production.

I think we could have waited with this. Maybe we can make the Landing page with something like "Under construction" and don't like the docs yet. cc @brijeshb42

**Pull request merged**